### PR TITLE
support OpenSplice 6.5

### PR DESCRIPTION
**Pull request by dirk-thomas**
@esteve @tfoote @wjwwood Please review.

### CODE REVIEW

File: rosidl_typesupport_opensplice_cpp/rosidl_typesupport_opensplice_cpp/rosidl_generator_dds_idl_extension.py [hunk 1]
```diff
@@ -30,10 +30,7 @@ def get_include_directives(spec, subfolders):
     # prepend directives to include opensplice specific idl file
     for field in spec.fields:
         if field.type.pkg_name == 'builtin_msgs':
-            includes[:0] = [
-                '#ifndef OSPL_IDL_COMPILER',
-                '#include <dds_dcps.idl>',
-                '#endif']
+            includes.insert(0, '#include <dds_dcps.idl>')
```
Comment (wjwwood) [inline rosidl_typesupport_opensplice_cpp/rosidl_typesupport_opensplice_cpp/rosidl_generator_dds_idl_extension.py]: What's the idea with this change? Why do you no longer need to check if that is defined?

File: opensplice_cmake_module/cmake/Modules/FindOpenSplice.cmake [hunk 2]
```diff
@@ -64,7 +64,6 @@ if(NOT "${_ospl_home} " STREQUAL " ")
     "${_ospl_home}/include/dcps/C++/SACPP")
   set(OpenSplice_LIBRARIES
     "cmxml"
-    "commonserv"
```
Comment (wjwwood) [inline opensplice_cmake_module/cmake/Modules/FindOpenSplice.cmake]: Have you tried this with older versions of OpenSplice? Do we care about that?

File: rosidl_typesupport_opensplice_cpp/rosidl_typesupport_opensplice_cpp/rosidl_generator_dds_idl_extension.py (hunk 1 above)
Comment (dirk-thomas) [inline rosidl_typesupport_opensplice_cpp/rosidl_typesupport_opensplice_cpp/rosidl_generator_dds_idl_extension.py]: The usage of `--dds-types` is deprecated and made the DDS builtin types magically available before.

The `ifndef` was a workaround for `idlpp` not finding the `dds_dcps.idl` file. But when correctly adding the template path as an include directory that is not necessary anymore.

Comment (wjwwood) [inline rosidl_typesupport_opensplice_cpp/rosidl_typesupport_opensplice_cpp/rosidl_generator_dds_idl_extension.py]: Ok, sounds good to me.

File: opensplice_cmake_module/cmake/Modules/FindOpenSplice.cmake (hunk 2 above)
Comment (dirk-thomas) [inline opensplice_cmake_module/cmake/Modules/FindOpenSplice.cmake]: Of course it still works with the current LGPL version of OpenSplice (6.4).

Afaik we don't use `commonserv`.

**Comment by wjwwood** [comment]
+1

**Comment by tfoote** [comment]
+1

**Comment by esteve** [comment]
+1

**Pull request merged**